api/v1alpha1/crds/caren.nutanix.com_awsclusterconfigs.yaml

```diff
@@ -298,6 +298,23 @@ spec:
                         type: string
                     type: object
                 type: object
+              encryption:
+                description: |-
+                  Encryption defines the configuration to enable encryption at REST
+                  This configuration is used by API server to encrypt data before storing it in ETCD.
+                  Currently the encryption only enabled for secrets and configmaps.
+                properties:
+                  providers:
+                    default: aescbc
+                    description: Encryption providers
+                    enum:
+                    - aescbc
+                    - aesgcm
```

Comment:
@dlipovetsky @dkoshkin the result of the SPIKE recommends that we should also allow `aesgcm` as one of the provider and default provider. https://jira.nutanix.com/browse/D2IQ-100160?focusedId=6347378&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-6347378

The `aesgcm` requires rotation of the keys every 200k writes. https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#providers
Given that we do not rotate keys, should we support it in CAREN API. 
If not, then it leaves only `aescbc` as a provider that we can support right now to support Konvoy usecase. 
We can keep the API flexible, I am wondering if we should allow `aesgcm` as one of the option without providing a way to rotate the keys before 200k writes.

Reply:
Good catch, IMO CAREN should not allow `aesgcm` since it would be too easy to have a vulnerable cluster over time, and should be secure by default. I think these kinds of details is another reason for CAREN to take care of token creation, and can extend it with rotation in the future.

Reply:
`aescbc` is also not recommended `due to CBC's vulnerability to padding oracle attacks`...

How about we support `aescbc` and `secretbox` as at least that doesn't seem to have any known vulnerabilities (yet)?

Reply:
The only way to `be secure by default` is to use KMS v2 right now i think, with `secretbox` a second place alternative.

Reply:
Note that OpenShift only supports `aescbc` and `aesgcm` (with weekly key rotation) - https://docs.openshift.com/container-platform/4.15/security/encrypting-etcd.html#etcd-encryption-types_encrypting-etcd.

Reply:
We can add `secretbox` provider. I will add it to the API.